pkg/riff/commands/namespace.go

```diff
@@ -153,3 +154,53 @@ func NamespaceCleanup(c *core.Client) *cobra.Command {
 
 	return command
 }
+
+func NamespaceAddCredentials(c *core.Client) *cobra.Command {
+	options := core.NamespaceAddCredentialsOptions{}
+
+	command := &cobra.Command{
+		Use:     "add-credentials",
+		Short:   "add secret and bind it to the " + env.Cli.Name + " service account",
+		Example: `  ` + env.Cli.Name + ` namespace add-credentials --secret build-secret --namespace default --docker-hub johndoe`,
+		Args:    cobra.ExactArgs(namespaceAddCredentialsNumberOfArgs),
+		PreRunE: FlagsValidatorAsCobraRunE(
+			FlagsValidationConjunction(
+				ValidDnsSubdomain("namespace"),
+				AtMostOneOf("gcr", "docker-hub", "registry-user"),
+				FlagsDependency(Set("registry-user"), NotBlank("registry")),
+				FlagsDependency(Set("registry"),
+					NotBlank("registry-user"),
+					SupportedRegistryProtocol(func() string {
+						return options.Registry
+					})),
+				NotBlank("secret"),
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := (*c).NamespaceAddCredentials(options); err != nil {
+				return err
+			}
+			PrintSuccessfulCompletion(cmd)
+			return nil
+		},
+	}
+
+	command.Flags().StringVar(&options.NamespaceName, "namespace", "", "the `namespace` of the credentials to be added")
```

Comment:
It seems odd that this uses a flag to pick the namespace, while `namespace init` uses an argument. I'd expect them to use the same pattern. (although I personally prefer using a flag)

Reply:
Another approach would be to move this into a top level command `riff credentials`

Reply:
Yep, there is a similar discussion going on in the PR. I'd argue for a separate top-level command.

Reply:
Done in newest commit.